core/Cargo.toml

```
@@ -46,7 +46,7 @@ ring = { version = "0.16.9", features = ["alloc", "std"], default-features = fal
 async-std = { version = "1.6.2", features = ["attributes"] }
 base64 = "0.13.0"
 criterion = "0.4"
-libp2p = { path = "..", features = ["full"] }
+libp2p = { path = "../meta/", features = ["full"] }
```

```suggestion
libp2p = { path = "../libp2p/", features = ["full"] }
```

Would it not make sense to name the folder like the crate name? Tokio seems to do just that. What do other projects do?

https://github.com/tokio-rs/tokio/tree/master/tokio

---

Would you rename all the other directories to their actual crate name too?

---

It might be a bit of repetition but I think I'd prefer just straight up using the crate name for all directories.

The directory hierarchy is also something I am not too sure about but that is orthogonal :)

---

> Would you rename all the other directories to their actual crate name too?

Given that each of them are sub-crates of `libp2p`, I think it is fine to keep them as is.